ledger/participant-state/kvutils/tools/integrity-check/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/LogAppendingCommitStrategySupport.scala

```diff
@@ -60,4 +68,28 @@ final class LogAppendingCommitStrategySupport()(implicit executionContext: Execu
         s"Expected: $expectedLogEntry${System.lineSeparator()}Actual: $actualLogEntry"
     )
   }
+
+  override def checkEntryIsReadable(keyBytes: Key, valueBytes: Value): Either[String, Unit] =
+    Envelope.open(valueBytes) match {
+      case Left(errorMessage) =>
+        Left(s"Invalid value envelope: $errorMessage")
+      case Right(Envelope.LogEntryMessage(logEntry)) =>
+        val _ = DamlLogEntryId.parseFrom(keyBytes)
+        if (logEntry.getPayloadCase == DamlLogEntry.PayloadCase.PAYLOAD_NOT_SET)
+          Left("Log entry payload not set.")
+        else
+          Right(())
+      case Right(Envelope.StateValueMessage(value)) =>
+        val key = DefaultStateKeySerializationStrategy.deserializeStateKey(keyBytes)
```

Comment:
You might want to inject the `StateKeySerializationStrategy` as well so that this function may be used with different ones (e.g., some implementations may choose to prefix state keys).

Reply:
Great point. Fixed. I'm using the one that's already part of the `Support`.